kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/HelmLintMojo.java

```diff
@@ -13,16 +13,43 @@
  */
 package org.eclipse.jkube.maven.plugin.mojo.build;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
 import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.eclipse.jkube.kit.resource.helm.HelmConfig;
+import org.eclipse.jkube.kit.resource.helm.HelmService;
 
 @Mojo(name = "helm-lint", defaultPhase = LifecyclePhase.INTEGRATION_TEST, requiresDependencyResolution = ResolutionScope.COMPILE)
 public class HelmLintMojo extends AbstractHelmMojo {
 
+  @Override
+  public void init() throws MojoFailureException {
+    super.init();
+
+    checkChartsExist(getHelm());
+  }
+
+  private void checkChartsExist(final HelmConfig helmConfig) {
+    for (HelmConfig.HelmType helmType : helmConfig.getTypes()) {
+      final Path chart = Paths.get(helmConfig.getOutputDir(), helmType.getOutputDir(), HelmService.CHART_FILENAME);
+      if (Files.notExists(chart)) {
+        logChartNotFoundWarning(chart);
+      }
+    }
+  }
+
   @Override
   public void executeInternal() throws MojoExecutionException {
     jkubeServiceHub.getHelmService().lint(getHelm());
   }
+
+  protected void logChartNotFoundWarning(final Path chart) {
+    getKitLogger().warn("No Helm chart has been generated yet by the k8s:helm goal at: " + chart);
```

Comment:
Can we use `getLogPrefix()` here so that we don't need to override method in OpenShift Maven Plugin just for changing prefix?

Reply:
We could, as `getLogPrefix()` would return the same as the goal prefix. But those two are not related.

The `getLogPrefix()` for "k8s" is defined in [`org.eclipse.jkube.maven.plugin.mojo.build.AbstractJKubeMojo.DEFAULT_LOG_PREFIX`](https://github.com/Jurrie/jkube/blob/ab0e10361a71635116eaa7c5db5a8ddf07745464/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractJKubeMojo.java#L62)) while the mojo prefix is defined in the [pom.xml](https://github.com/Jurrie/jkube/blob/ab0e10361a71635116eaa7c5db5a8ddf07745464/kubernetes-maven-plugin/plugin/pom.xml#L181). There is nothing that relates the two, other than the values being the same.

Same goes for "oc": the log prefix is defined in [`org.eclipse.jkube.maven.plugin.mojo.OpenShift.DEFAULT_LOG_PREFIX`](https://github.com/Jurrie/jkube/blob/ab0e10361a71635116eaa7c5db5a8ddf07745464/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/OpenShift.java#L23) and the mojo prefix in the [pom.xml](https://github.com/Jurrie/jkube/blob/ab0e10361a71635116eaa7c5db5a8ddf07745464/openshift-maven-plugin/plugin/pom.xml#L70).

Because nothing is guaranteeing that those values are (and stay) the same, I did not use `getLogPrefix()`.

Reply:
> Because nothing is guaranteeing that those values are (and stay) the same

Sorry, I didn't understand this correctly. Could you please elaborate on this?

Reply:
For example: I can change [the logger prefix](https://github.com/Jurrie/jkube/blob/ab0e10361a71635116eaa7c5db5a8ddf07745464/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/OpenShift.java#L23) to read "ocp" instead of "oc". Then the Maven goal would still be "oc:helm"; it would not change to "ocp:helm". Also no error would occur, and no unit test would fail. That's what I mean with "nothing is guaranteeing that those values are (and stay) the same."

Because of this, you can not assume `getLogPrefix()` will return the same value as the Maven mojo prefix.

Reply:
Thanks for clarifying!